src/Simulation/Simulation.jl

```diff
@@ -80,27 +80,32 @@ Base.convert(T::Type{NamedTuple}, x::Simulation) = T(x)
 
 function Simulation(nt::NamedTuple)
     missing_tuple = NamedTuple(missing)
-    epot = ElectricPotential(nt.electric_potential)
-    T = eltype(epot.data)
-    sim = Simulation{T}( Dict(nt.detector_json_string) )
-    sim.electric_potential = epot
-    sim.q_eff_imp = EffectiveChargeDensity(nt.q_eff_imp)
-    sim.q_eff_fix = EffectiveChargeDensity(nt.q_eff_fix)
-    sim.ϵ_r = DielectricDistribution(nt.ϵ_r)
-    sim.point_types = PointTypes(nt.point_types)
-    sim.imp_scale = if !haskey(nt, :imp_scale) 
-        @warn """Stored simulation does not have a field for `imp_scale` (impurity scale) as this was 
-        first introduced in SolidStateDetectors.jl v0.8 for improved depletion handling.
-        It is advised to recalculate the simulation with the latest version.
-        The field `imp_scale` is determined from `point_types`: 
-            * undepleted point -> imp_scale = 0;  
-            * depleted point -> imp_scale = 1;  
-        """
-        ImpurityScale(T.(.!is_undepleted_point_type.(sim.point_types.data)), sim.point_types.grid)
+    if nt.electric_potential !== missing_tuple
+        epot = ElectricPotential(nt.electric_potential)
+        T = eltype(epot.data)
+        sim = Simulation{T}( Dict(nt.detector_json_string) )
+        sim.electric_potential = epot
+        sim.q_eff_imp = EffectiveChargeDensity(nt.q_eff_imp)
+        sim.q_eff_fix = EffectiveChargeDensity(nt.q_eff_fix)
+        sim.ϵ_r = DielectricDistribution(nt.ϵ_r)
+        sim.point_types = PointTypes(nt.point_types)
+        sim.imp_scale = if !haskey(nt, :imp_scale) 
+            @warn """Stored simulation does not have a field for `imp_scale` (impurity scale) as this was 
+            first introduced in SolidStateDetectors.jl v0.8 for improved depletion handling.
+            It is advised to recalculate the simulation with the latest version.
+            The field `imp_scale` is determined from `point_types`: 
+                * undepleted point -> imp_scale = 0;  
+                * depleted point -> imp_scale = 1;  
+            """
+            ImpurityScale(T.(.!is_undepleted_point_type.(sim.point_types.data)), sim.point_types.grid)
+        else
+            ImpurityScale(nt.imp_scale)
+        end
+        sim.electric_field = haskey(nt, :electric_field) && nt.electric_field !== missing_tuple ? ElectricField(nt.electric_field) : missing
     else
-        ImpurityScale(nt.imp_scale)
+        T = Float32
```

Comment:
Unrelated to depletion voltages but should be make `T` a keyword argument of this function rather than a hard coded variable in the inside of the function? What if the user wants the `Simulation` in `Float64`.

Reply:
Ups, sorry. Remaining of Copy&Paste from testing script.